UnityProject/Assets/Scripts/Core/Chat/ChatRelay.cs

```diff
@@ -304,29 +304,34 @@ public void AddMentorPrivMessageToClient(string message)
 
 		switch (channels)
```

Comment:
It should be noted that the switch doesn't understand combinations of channels. I'm not sure if that can happen with the listed ones, can anyone else confirm that it doesn't?

Reply:
```cs
switch (channels)
{
    case var c when c.HasFlag(ChatChannel.Common):
        DoCommonThing();
        break;
    case var c when c.HasFlag(ChatChannel.Action):
        DoActionThing();
        break;
    ...
}
```

Something like that should work. I imagine you could use the fancier switch expressions as well

Reply:
there we go